bin/mod5_diffusion.py

```diff
@@ -27,15 +27,13 @@ def diffusion(nn, y, alpha=0.5, max_iter=100):
     """
     alpha = 0.5
     max_iter = 100
-    y = y.to_numpy()
+    y = y.to_numpy().astype('float32')
     F = y
     ## initialize an empty array for the heat
-    Fs = np.empty((nn.shape[0], 0), float)
-    ## restart
-    fY = (1 - alpha) * y
+    Fs = np.empty((nn.shape[0], 0), 'float32')
     for i in range(0, max_iter):
         Fs = np.append(Fs, F, axis=1)
-        F = alpha * nn @ Fs[:, [i]] + fY
+        F = nn @ (alpha * Fs[:, [i]]) + (1 - alpha) * y
```

Comment:
any benefit for using `(1 - alpha) * y` instead of using `fY`?

Reply:
> any benefit for using `(1 - alpha) * y` instead of using `fY`?

I thought it adds intuitiveness to the equation, and no drawbacks on speed.